assets/content/cookbook/Graphics/Creating-SVG/01.Drawing-a-Circle.md

```diff
@@ -0,0 +1,313 @@
+```
+
+## Setting up SVG
+
+To create SVG we create a root node which we append into our HTML document.
+```haxe 
+// create SVGElement
+var svgElement: SVGElement = cast Browser.document.createElementNS("http://www.w3.org/2000/svg", 'svg');
+// Append to the document body
+Browser.document.body.appendChild(svgElement);
+```
+With SVG or canvas we need to setup the style so that coordinates are relative to the top corner.
+This is done using the 'style' property of the svg node.
+```haxe
+var style:CSSStyleDeclaration = element.style;
+style.paddingLeft = "0px";
+style.paddingTop = "0px";
+style.left = "0px";
+style.top = "0px";
+style.position = "absolute";
+```
+It is important to set the width and height attributes, those can be set like this:
+```haxe
+svgElement.setAttribute('width', '800');
+svgElement.setAttribute('height', '600');
+```
+But really this is ugly, it might be ok for JavaScript developers setting attributes like this but it's pretty inflexible if we want to tween or animate attributes.
+So instead we can wrap the 'set' and 'get' attribute methods for width and height, so for width we can create a getter/setter.
+```haxe
+    public var width(get, set):Int;
+    inline public function set_width(value:Int):Int {
+        // pass in a Integer width value and convert it to a String for the setAttribute method.
+        this.setAttribute("width", Std.string(value));
+        return value;
+    }
+    inline public function get_width():Int {
+        // convert String value back an Integer for easier manipulation.
+        return Std.parseInt(this.getAttribute("width"));
+    }
+```
+if we wrap the SVGElement in an abstract we can then call these properties directly on the abstract. This allows much easier manipulation of the SVG node's width.
+```haxe
+svgRootAbstract.width = 800;
+```
+
+## The full abstract for our root SVG element
+
+```haxe
+import js.Browser;
+import js.html.svg.SVGElement;
+import js.html.Element;
+import js.html.CSSStyleDeclaration;
+import js.html.Node;
+
+abstract SvgRoot(SVGElement) from SVGElement to SVGElement {
+  inline public static var svgNameSpace:String = "http://www.w3.org/2000/svg" ;
+  inline public function new(?e:SVGElement) {
+    if( e == null) {
+      this = create();
+    } else {
+      this = e;
+    }
+  }
+ 
+  inline static public function create():SvgRoot {
+    var svgElement:SVGElement = cast Browser.document.createElementNS(svgNameSpace, 'svg');
+    var element:Element = cast svgElement;
+    var style:CSSStyleDeclaration = element.style;
+    style.paddingLeft = "0px";
+    style.paddingTop = "0px";
+    style.left = "0px";
+    style.top = "0px";
+    style.position = "absolute";
+    Browser.document.body.appendChild( element );
+    var svgRoot:SvgRoot = svgElement;
+    return svgRoot;
+  }
+  
+  // allows us to add children svg elements 
+  inline public function add(element:SVGElement):Void {
+    var node:Node = cast element;
```

Comment:
Is this cast really needed? SVGElement extends GraphicsElement › Element › DOMElement › Node › EventTarget